### DISCUSSION TITLE
Document Library Edit new menu

### DISCUSSION THREAD

**Discussion by matjazreberc** [pnp/pnpcore#1242] [Q&A] [answered]
Hello,

is it possible to Programmatically change the New Menu in SharePoint Online using PnPCore SDK, 
so that also Excel and others would be selected?

![image](https://github.com/pnp/pnpcore/assets/60425382/62c07600-9a4b-4b7a-acc3-1cbccb41c502)

Thank you, Matjaz Reberc

**Comment by jansenbe** [comment, marked as answer]
@matjazreberc : did not try this myself yet, but please check and update the `IView.NewDocumentTemplates` property and see if that works for you.

**Reply by matjazreberc** [reply]
Hello @jansenbe ,

I did work.
Here is my code if some one needs it:


```
var newMenu = myListViews.Views.AsRequested().FirstOrDefault(p => p.Title == "All Documents") ;
                newMenu.NewDocumentTemplates = "[{\"templateId\":\"NewFolder\",\"title\":\"Folder\",\"visible\":true}," +
                                                         "{\"templateId\":\"NewDOC\",\"title\":\"Word document\",\"visible\":true}," +
                                                         "{\"templateId\":\"NewXSL\",\"title\":\"Excel workbook\",\"visible\":true}," +
                                                         "{\"templateId\":\"NewPPT\",\"title\":\"PowerPoint presentation\",\"visible\":true}," +
                                                         "{\"templateId\":\"NewONE\",\"title\":\"OneNote notebook\",\"visible\":true}," +
                                                         "{\"templateId\":\"NewVSDX\",\"title\":\"Visio drawing\",\"visible\":true}]";

                await newMenu.UpdateAsync();
````

I also fount this, [https://techcommunity.microsoft.com/t5/sharepoint-developer/programatically-change-the-new-menu-in-modern-libraries/m-p/1033286](url)

Thank you, @jansenbe
